test/generate.test.js

```diff
@@ -0,0 +1,18 @@
+const { execSync } = require('child_process');
+const fs = require('fs');
+
+test('CLI generates correct SVG file', () => {
+
+  try {
+    const output = execSync(`node --experimental-specifier-resolution=node dist/index.js -p test/plan.json -o .`);
+
+    expect(output.toString()).toContain('SVG file saved successfully');
+
+    const outputSvgContent = fs.readFileSync('plan.svg', 'utf-8');
+    const expectedSvgContent = fs.readFileSync('test/expected.svg', 'utf-8');
+    expect(outputSvgContent).toEqual(expectedSvgContent);
```

Comment:
I am not sure if this should be the way to test whether the tool generates the correct SVG file. Is there any better way for this?

Reply:
An alternative would be to check that the file is there and then check its contents (which you could do with a string in a test file somewhere).  But the input is coming from a separate test file.   It seems appropriate to have a separate golden file to match.  This could be changed to look for test files and compare them to golden ones (allowing it to check for multiple correct files) instead of being limited to a single file.

Reply:
okay, so what I understand is you suggest having multiple tests verifying the correctness of the tool instead of just limiting it to the one plan and its expected SVG. Is that correct?

Reply:
Long term that's the way to go.  But for now this is a step in that direction.